### DISCUSSION TITLE
Widget: How to have 3 in the same row?

### DISCUSSION THREAD

**Discussion by realtebo** [Laravel-Backpack/community-forum#631] [Q&A (Help)] [answered]
I thinks it's another lack of documentation ... sigh ...

I am doing this
      
      @extends(backpack_view('blank'))
      
      @php
              $classi_totali = \App\Models\InstituteClass::query()->count();
              $classi_senza_insegnanti =\App\Models\InstituteClass::query()->doesntHave('teachers')->count();
      
              $widgets['before_content'][] = [
                      'type'        => 'progress_white',
                      'class'       => 'card mb-2',
                      'value'       => $classi_totali,
                      'description' => 'Classi',
                      'progress'    => 100- intval(100 * $classi_senza_insegnanti / $classi_totali), // integer
                      'hint'        => $classi_senza_insegnanti .' senza insegnanti.',
              ];
      
      
              $classi_senza_studenti =\App\Models\InstituteClass::query()->doesntHave('teachers')->count();
      
              $widgets['before_content'][] = [
                      'type'        => 'progress_white',
                      'class'       => 'card mb-2',
                      'value'       => $classi_totali,
                      'description' => 'classi',
                      'progress'    => 100- intval(100 * $classi_senza_studenti / $classi_totali), // integer
                      'hint'        => $classi_senza_studenti .' senza STUDENTI.',
              ];
      
              $insegnanti_totali = \App\Models\User::byRole(\App\Models\User::ROLE_TEACHER)->count();
              $insegnanti_senza_classi =\App\Models\User::byRole(\App\Models\User::ROLE_TEACHER)->doesntHave('teacher_classes')->count();
      
              $widgets['before_content'][] = [
                      'type'        => 'progress_white',
                      'class'       => 'card mb-2',
                      'value'       => $insegnanti_totali,
                      'description' => 'insegnanti',
                      'progress'    => 100- intval(100 * $insegnanti_senza_classi / $insegnanti_totali), // integer
                      'hint'        => $insegnanti_senza_classi .' senza classi.',
              ];
      
      @endphp
      
      
      @section('content')
          <p>Your custom HTML can live here</p>
      @endsection

Getting this

![image](https://github.com/Laravel-Backpack/community-forum/assets/1268699/5db78c3a-6b7f-42df-b457-dc7c8a986b38)

**Comment by realtebo** [comment]
After looking into source code ...

      @php
      
              $classi_totali = \App\Models\InstituteClass::query()->count();
              $classi_senza_insegnanti =\App\Models\InstituteClass::query()->doesntHave('teachers')->count();
              $classi_senza_studenti =\App\Models\InstituteClass::query()->doesntHave('teachers')->count();
              $insegnanti_totali = \App\Models\User::byRole(\App\Models\User::ROLE_TEACHER)->count();
              $insegnanti_senza_classi =\App\Models\User::byRole(\App\Models\User::ROLE_TEACHER)->doesntHave('teacher_classes')->count();
      
      
              $widgets['before_content'][] = [
                  'type' => 'div',
                  'class' => 'row',
                  'content' => [
                      [
                          'type'        => 'progress_white',
                          'class'       => 'card mb-2',
                          'value'       => $classi_totali,
                          'description' => 'Classi',
                          'progress'    => 100- intval(100 * $classi_senza_insegnanti / $classi_totali), // integer
                          'hint'        => $classi_senza_insegnanti .' senza insegnanti.',
                      ],
                      [
                          'type'        => 'progress_white',
                          'class'       => 'card mb-2',
                          'value'       => $classi_totali,
                          'description' => 'classi',
                          'progress'    => 100- intval(100 * $classi_senza_studenti / $classi_totali), // integer
                          'hint'        => $classi_senza_studenti .' senza STUDENTI.',
                      ],
                      [
                          'type'        => 'progress_white',
                          'class'       => 'card mb-2',
                          'value'       => $insegnanti_totali,
                          'description' => 'insegnanti',
                          'progress'    => 100- intval(100 * $insegnanti_senza_classi / $insegnanti_totali), // integer
                          'hint'        => $insegnanti_senza_classi .' senza classi.',
                      ],
                      [
                          'type'        => 'progress_white',
                          'class'       => 'card mb-2',
                          'value'       => $insegnanti_totali,
                          'description' => 'insegnanti',
                          'progress'    => 100- intval(100 * $insegnanti_senza_classi / $insegnanti_totali), // integer
                          'hint'        => $insegnanti_senza_classi .' senza classi.',
                      ]
                  ]
              ];
      @endphp

**Comment by realtebo** [comment]
Do you really suppose we understand that we must have a widget of type div with a content that is an array of widgets?

**Comment by o15a3d4l11s2** [comment, marked as answer]
The documentation for the `div` widget is here: https://backpackforlaravel.com/docs/6.x/base-widgets#div-1 and it states
> Allows you to include multiple widgets within a "div" element with the attributes of your choice. For example, you can include multiple widgets within a \<div class="row">\</div> with the code below

Sorry for any confusion. Feel free to share your feedback and ideas how the documentation can be improved to make it easier to find.
